improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -164,321 +280,104 @@ def calculate_d_enthalpy_dt(mixing_ratio, specific_heat,
+        Calculate an array of wet bulb temperatures from inputs in
+        the correct units.
 
-        Args:
-            cube (numpy.ndarray):
-                Array of temperature.
-
-            low (int or float):
-                Lowest allowable temperature for check
-
-            high (int or float):
-                Highest allowable temperature for check
-
-        Raises:
-            UserWarning : If any of the values in cube.data are outside the
-                          bounds set by the low and high variables.
-        """
-        if cube.max() > high or cube.min() < low:
-            emsg = ("Wet bulb temperatures are being calculated for conditions"
-                    " beyond the valid range of the saturated vapour pressure"
-                    " lookup table (< {}K or > {}K). Input cube has\n"
-                    "Lowest temperature = {}\nHighest temperature = {}")
-            warnings.warn(emsg.format(low, high, cube.min(),
-                                      cube.max()))
-
-    def lookup_svp(self, temperature):
-        """
-        Looks up a value for the saturation vapour pressure of water vapour
-        using the temperature and a table of values. These tabulated values
-        have been calculated using the utilities.ancillary_creation
-        SaturatedVapourPressureTable plugin that uses the Goff-Gratch method.
+        A Newton iterator is used to minimise the gradient of enthalpy
+        against temperature.
 
         Args:
+            pressure (numpy.ndarray):
+                Array of air Pressure (Pa).
+            relative_humidity (numpy.ndarray):
+                Array of relative humidities (1).
             temperature (numpy.ndarray):
-                Array of air temperatures (K).
+                Array of air temperature (K).
+
         Returns:
             numpy.ndarray:
-                Array of saturated vapour pressures (Pa).
-        """
-        # We subtract T_INCREMENT from T_MAX to get the upper bound to which we
-        # clip input temperatures. This ensures that we do not attempt an
-        # interpolation that requires a value beyond the SVP table maximum.
-        T_max = svp_table.T_MAX - svp_table.T_INCREMENT
-        T_min = svp_table.T_MIN
-        delta_T = svp_table.T_INCREMENT
-        self.check_range(temperature, T_min, T_max)
-        T_clipped = np.clip(temperature, T_min, T_max)
-
-        # Note the indexing below differs by -1 compared with the UM due to
-        # Python vs. Fortran indexing.
-        table_position = (T_clipped - T_min + delta_T)/delta_T - 1.
-        table_index = table_position.astype(int)
-        interpolation_factor = table_position - table_index
-        return ((1.0 - interpolation_factor) * svp_table.DATA[table_index] +
-                interpolation_factor * svp_table.DATA[table_index + 1])
+                Array of wet bulb temperature (K).
 
-    @staticmethod
-    def pressure_correct_svp(svp, temperature, pressure):
```

Comment:
Made utility `calculate_svp_in_air`, which is imported from a couple of other places.